home/modules/secureEnv/onePassword.nix

```diff
@@ -17,21 +17,45 @@ let
     else
       ''${pkgs.libsecret}/bin/secret-tool lookup namespace '${namespace}' key '${key}' '';
 
+  storeSshKeyCmd = keyfile: value:
+    ''echo "${value}" | ${pkgs.openssl}/bin/openssl pkcs8 -topk8 -nocrypt >${keyfile} && \
+      chmod 600 ${keyfile} && \
+      ssh-add ${keyfile} && \
```

Comment:
Can't we pipe into `ssh-add` instead of using tmp files?

Something like:

```bash
echo "${value}" | ${pkgs.openssl}/bin/openssl pkcs8 -topk8 -nocrypt >${keyfile} | ssh-add -
```

or maybe
```bash
ssh-add <(echo "${value}" | ${pkgs.openssl}/bin/openssl pkcs8 -topk8 -nocrypt >${keyfile})
```

Not sure about the exact syntax, but there should be a way ;)

Reply:
Not using a temp file was what I really wanted, and is clearly better.  I have given up on the comment (inferred from the filename), seems a small loss to avoid the ugly temp file.  Thanks for poking me about this! :grin: